Source/JavaScriptCore/dfg/DFGIntegerRangeOptimizationPhase.cpp

```diff
@@ -1920,12 +1925,62 @@ class IntegerRangeOptimizationPhase : public Phase {
 
     unsigned m_iterations { 0 };
 };
-    
+
 } // anonymous namespace
 
+// These functions are useful for writing tests for this phase.
+void startDebugRecordingIRO()
+{
+    RELEASE_ASSERT(!Options::useConcurrentJIT());
+    RELEASE_ASSERT(Options::useDollarVM());
```

Comment:
Why is this assert helpful?

Reply:
We should ensure that debugging functions aren't abused in production code.  That said, please use `DollarVMAssertScope assertScope;` instead.